ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/PayloadAttributesCalculator.java

```diff
@@ -82,25 +92,47 @@ public void updateProposers(
     final UInt64 epoch = spec.computeEpochAtSlot(blockSlot);
     return getStateInEpoch(epoch)
         .thenApplyAsync(
-            maybeState -> calculatePayloadAttributes(blockSlot, epoch, maybeState), eventThread);
+            maybeState ->
+                calculatePayloadAttributes(
+                    blockSlot, epoch, maybeState, mandatoryPayloadAttributes),
+            eventThread);
   }
 
   private Optional<PayloadAttributes> calculatePayloadAttributes(
-      final UInt64 blockSlot, final UInt64 epoch, final Optional<BeaconState> maybeState) {
+      final UInt64 blockSlot,
+      final UInt64 epoch,
+      final Optional<BeaconState> maybeState,
+      final boolean mandatoryPayloadAttributes) {
     eventThread.checkOnEventThread();
     if (maybeState.isEmpty()) {
       return Optional.empty();
     }
     final BeaconState state = maybeState.get();
     final UInt64 proposerIndex = UInt64.valueOf(spec.getBeaconProposerIndex(state, blockSlot));
     final ProposerInfo proposerInfo = proposerInfoByValidatorIndex.get(proposerIndex);
-    if (proposerInfo == null) {
+    if (proposerInfo == null && !mandatoryPayloadAttributes) {
       // Proposer is not one of our validators. No need to propose a block.
       return Optional.empty();
     }
     final UInt64 timestamp = spec.computeTimeAtSlot(state, blockSlot);
     final Bytes32 random = spec.getRandaoMix(state, epoch);
-    return Optional.of(new PayloadAttributes(timestamp, random, proposerInfo.feeRecipient));
+    return Optional.of(new PayloadAttributes(timestamp, random, getFeeRecipient(proposerInfo)));
+  }
+
+  // this function MUST return a fee recipient.
+  private Bytes20 getFeeRecipient(final ProposerInfo proposerInfo) {
+    if (proposerInfo != null) {
+      return proposerInfo.feeRecipient;
+    }
+    if (proposerDefaultFeeRecipient.isPresent()) {
+      LOG.warn(
+          "Payload Attributes are required but current proposer hasn't prepared. Using Beacon Node's default values.");
+      return proposerDefaultFeeRecipient.get();
+    }
+    // TODO: rise a RED EVENT LOG?
+    LOG.error(
+        "Payload Attributes are required but current proposer hasn't prepared and there is no default fee recipient configured. FEES ARE GOING TO BE BURNED!");
+    return Bytes20.ZERO;
```

Comment:
I suspect at this point failing to produce the block is probably the better approach. That way the error will propagate out to the validator client whereas this error will only show up in the beacon node logs which may not be in the VC user's control.

Actually speaking of that we need to think through how to handle these errors/warnings for people like Infura - they get worried when there are a heap of error messages in their logs but things like this may be triggered by external VC activities and not necessarily indicate a problem with the beacon node.  Simplest answer is to push any of this through `ValidatorLogger` since we have a CLI option to disable those logs.

Reply:
OK, yes.
The usage of the default depends by the VC behaviour. But if the default is not configured is a bad configuration of the BN, but we are already noisy on that each time `prepare_beacon_proposer` is called.